system/include/emscripten/threading.h

```diff
@@ -48,6 +48,11 @@ int emscripten_is_main_runtime_thread(void);
 // this returns 1.
 int emscripten_is_main_browser_thread(void);
 
+// Return non-zero if the calling thread supports Atomic.wait (For example if called from
+// the main browser or audio worklet thread, this function will return zero since blocking
+// is not allowed there).
+int emscripten_thread_supports_atomics_wait(void);
```

Comment:
Can we avoid adding the public API here?   Its seems like keeping this internal would make this change much smaller.   

If you want to keep using it in the test code specifically you could perhaps forward declare the internal function there rather then making it a part of the public API.

If we do want to make this into a new public API then perhaps that can/should be its own standalone change.

Reply:
It’s more than the test, it reenables some form of locks for audio worklets. The test is to replace the existing one that was broken (and not being run).

More details in #22962.

Reply:
Ah I see.  Perhaps the PR title and description could be improved then.   The title sounds like this change is about testing, but there is a function change in there.   I guess "testing" is ambiguous in this context.

Reply:
Title and description changed (I'll make the API changes tomorrow).

Reply:
What do you think about keeping the public API constant for this change?   Do we need to make this new API public?

Reply:
I'm considering moving the call back to being private, as you suggested, and remove it from the test. I had a think about it:

- Audio worklets don't support atomic wait, so the result is a given
- I can't think of a use case where I would test for wait support then run another branch (real use cases, since the AW code is run in a worklet)
- We know we're in an AW from the rest of the API calls

I'll look in the morning.

Reply:
If you want you can still forward declare the private function in the test code... its up to you.